ui/src/app/pages/workspace/workspace-edit.spec.tsx

```diff
@@ -30,13 +30,20 @@ import {cdrVersionStore, profileStore, serverConfigStore} from 'app/utils/stores
 import {AccessTierShortNames} from 'app/utils/access-tiers';
 import {navigateSpy} from 'testing/navigation-mock';
 import {ProfileApiStub} from "testing/stubs/profile-api-stub";
+import {Dropdown} from "primereact/dropdown";
+
+import * as Authentication from "../../utils/authentication";
```

Comment:
our normal pattern is to just import what you use. The `*` import is interesting though - similar to a package import. The only thing I don't like about this is that it introduces this `Authentication` alias that doesn't truly correspond to anything. I would probably just switch to importing the specific functions for now

Reply:
I can not make mock work if import the function. See line 101, 102:
```
mockHasBillingScope = jest.spyOn(Authentication, 'hasBillingScope');
mockEnsureBillingScope = jest.spyOn(Authentication, 'ensureBillingScope');
```
what shall I put in Authentication field to make that work? 
I got this approach from: https://stackoverflow.com/questions/54245654/how-to-spy-on-a-default-exported-function-with-jest

Reply:
I see, this is OK then for now. In the future we may want to refactor the auth package anyways. It may be possible to stub it out like this as well though - I don't really have a strong prefrence between these two approaches:

https://github.com/all-of-us/workbench/blob/master/ui/src/app/components/app-router.spec.tsx#L8-L16

Reply:
I think I tried that(line the line 40-42 `workbench-gapi-client`). But having jest.mock in class level, I am not able to dynamic change returned value and do spy in different test case. I found the solution online, that is the only why I figured out I can use `mockHasBillingScope.mockImplementation` to change the returned value in different test case...